webapp/src/components/common/datePicker/index.tsx

```diff
@@ -14,25 +11,28 @@ import {
     useFloating,
     useInteractions,
 } from '@floating-ui/react';
-
+import React, {useCallback, useRef, useState} from 'react';
 import {DayPicker} from 'react-day-picker';
 
+import './style.scss';
+
 export type Props = {
-    value?: Date
-    children: React.ReactNode
-    onSelect: (date: Date) => void
-    closeOnSelect?: boolean
+    value?: Date;
+    children: React.ReactNode;
+    onSelect: (date: Date) => void;
+    closeOnSelect?: boolean;
 };
 
 const DatePicker = ({value, children, onSelect, closeOnSelect}: Props) => {
     const [isPopupOpen, setPopupOpen] = useState<boolean>(false);
+    const currentDate = useRef<Date>(new Date());
 
     const onSelectHandler = useCallback((day: Date) => {
         onSelect(day);
 
         // Checking it this way to handle null and undefined.
         // The default behavior is to close on select.
-        if (closeOnSelect !== false) {
+        if (closeOnSelect || typeof closeOnSelect === 'undefined') {
```

Comment:
Just a thought - as a simplification can we rework this to work off a positive case instead, so you could instead write something like `if (dontCloseOnSelect)`

Reply:
Fixed by adding a default value where I'm expanding the props. So I don't get null and undefined states here, only true or false. So this is now simplified to just `if (dontCloseOnSelect)`.